### DISCUSSION TITLE
Is it possible to inject an Entities Repository into the Entity class itself?

### DISCUSSION THREAD

**Discussion by janniks** [mikro-orm/mikro-orm#1512] [Q&A] [answered]
Would it somehow be possible to inject the repository methods of an entity into the entity class itself (as static methods)?

e.g. a `Book` class annotated as usual
But then somehow inject the repository into the class, so we could do the following in a more ActiveRecord style:
`Book.findOne(...)`

Is this somehow possible with mikro-orm, maybe using injection?

**Comment by B4nan** [comment, marked as answer]
Not really, this is data mapper, there are multiple entity managers, at least one for each request. What you can do is leverage `RequestContext` helper - you can get the EM from it statically (but only inside the handler).
